mitmproxy/addons/termlog.py

```diff
@@ -5,14 +5,19 @@
 
 
 class TermLog:
-    def __init__(self, outfile=sys.stdout):
+    def __init__(self, outfile=None):
         self.options = None
         self.outfile = outfile
 
     def configure(self, options, updated):
         self.options = options
 
     def log(self, e):
+        if self.outfile == None:
+            if log.log_tier(e.level) == log.log_tier("error"):
```

Comment:
I think this should be evaluated every time - not just on the first log message - because in the next line you set `self.outfile` to something.